ui/src/components/layouts/TheSignUp.vue

```diff
@@ -235,7 +213,126 @@
 </style>
 
 <script>
+import axios from "axios";
+import router from "../../router";
+
 export default {
+	name: "SignUp",
+	props: {
+		textMuteFirstName: { type: Boolean, default: true },
+		textMuteLastName: { type: Boolean, default: true },
+		textMuteCompanyName: { type: Boolean, default: true },
+		textMuteUserName: { type: Boolean, default: true },
+		textCheckUserName: { type: Boolean, default: true },
+		textMutePassword: { type: Boolean, default: true },
+	},
+	data() {
+		return {
+			msg: "",
+			msgCheck: "",
+			user: { type: Object },
+		}
+	},
 
+	methods: {
+		/**
+		 * validate input null
+		 * CreatedBy: DucLM (20/09/2021)
+		 */
+		validate() {
+			this.msg = "Không được để trống"
+			if (this.user.firstName == "" || this.user.firstName == null) {
+				this.textMuteFirstName = false;
+				this.$refs.firstName.focus();
+			} else {
+				this.textMuteFirstName = true;
+			}
+			if (this.user.lastName == "" || this.user.lastName == null) {
+				this.textMuteLastName = false;
+				this.$refs.lastName.focus();
+			} else {
+				this.textMutelastName = true;
+			}
+			if (this.user.companyName == "" || this.user.companyName == null) {
+				this.textMuteCompanyName = false;
+				this.$refs.companyName.focus();
+			} else {
+				this.textMuteCompanyName = true;
+			}
+			if (this.user.username == "" || this.user.username == null) {
+				this.textMuteUsername = false;
+				this.$refs.username.focus();
+			} else {
+				this.textMuteUsername = true;
+			}
+			if (this.user.password == "" || this.user.password == null) {
+				this.textMutePassword = false;
+				this.$refs.password.focus();
+			} else {
+				this.textMutePassword = true;
+				return true;
+			}
+			return false;
+		},
+		/**
+		 * click sign up
+		 * CreatedBy: DucLM (20/09/2021)
+		 */
+		btnClickSignUp() {
+			if (this.validate()) {
+				if (this.checkUsernameExist()) {
+					axios.post("http://localhost:3000/users", {
+						firstName: this.user.firstName,
```

Comment:
Gửi thẳng this.user không ngắn hơn ak em mà phải chia nhỏ kiểu vầy thế

Reply:
qua lúc em check gửi thẳng this.user nhưng nó không nhận nên đành phải làm thế, để nay em thử lại (chắc qua do máy em mất mạng nên nó chưa nhận@@)

Reply:
Em thay cục object kia thành this.data là dc mà. Nếu k dc thì mình thử debug xem tại sao. truyền lên v thì data server nó nhận dc đang ntn

Reply:
chỗ này em sửa được rồi ạ. em commit ở dưới